software/web/src/Client/midi-converter.js

```diff
@@ -0,0 +1,136 @@
+const midi = require('midi-file');
+const uuid = require('uuid');
+
+const NOTE_NAMES = [
+    "A",
+    "AS",
+    "B",
+    "C",
+    "CS",
+    "D",
+    "DS",
+    "E",
+    "F",
+    "FS",
+    "G",
+    "GS"
+];
+
+const NOTE_ON = "noteOn";
+const NOTE_OFF = "noteOff";
+
+const DEFAULT_NEW_CONFIG_NAME = "New Lighting Config";
+const DEFAULT_NEW_CONFIG_PATTERNS = [
+    {
+        "id": "green",
+        "leds": [
+          "#00FF00"
+        ],
+        "repeat": 0,
+        "offset": 0
+    }
+];
+const DEFAULT_NEW_CONFIG_COMMANDS = [
+    {
+        "command": "display",
+        "id": "green"
+    }
+];
+const CREATING_NEW_CONFIG = "No current song input. Creating new song template.";
+const ERROR_NEW_CONFIG_MUSIC = "Error parsing file - are you sure the file you provided is MIDI?";
+
+var getClaraNoteNameFromMidiNumber = (midiNumber) => {
+    return "" + NOTE_NAMES[((midiNumber - 21) % 12)] + (Math.floor((midiNumber - 24) / 12))
```

Comment:
No check to see if midiNumber is in range

Reply:
What is the range?